src/redux/features/accessions/accessionsThunks.ts

```diff
@@ -0,0 +1,66 @@
+import { Dispatch } from 'redux';
+import { RootState } from 'src/redux/rootReducer';
+import { SearchNodePayload } from 'src/types/Search';
+import strings from 'src/strings';
+import SeedBankService from 'src/services/SeedBankService';
+import { setAccessionsAction } from './accessionsSlice';
+
+const SEARCH_FIELDS_ACCESSIONS = ['id', 'accessionNumber', 'speciesName'];
+
+export type SearchResponseAccession = {
+  id: string;
+  accessionNumber: string;
+  speciesName: string;
+};
+
+export const requestAccessions = (organizationId: number, speciesId?: number) => {
+  return async (dispatch: Dispatch, _getState: () => RootState) => {
+    const stateSpeciesId = speciesId ?? 0;
+    const orgIdSpeciesId = `${organizationId}-${stateSpeciesId}`;
+
+    try {
+      const searchCriteria: { [key: string]: SearchNodePayload } = {};
+      searchCriteria.excludeUsedUp = {
+        operation: 'not',
+        child: {
+          operation: 'field',
+          field: 'state',
+          type: 'Exact',
+          values: [strings.USED_UP],
+        },
+      };
+
+      if (stateSpeciesId) {
```

Comment:
We have a similar use-case in planting sites, we use `-1` to retrieve all sites versus by specific id.